assets/src/edit-story/components/panels/link/index.js

```diff
@@ -99,15 +101,33 @@ function LinkPanel({ selectedElements, pushUpdateForObject }) {
     if (!isValidUrl(urlWithProtocol)) {
       return;
     }
+    const invalidLinkMessage = __('This is an invalid link!', 'web-stories');
     setFetchingMetadata(true);
     getLinkMetadata(urlWithProtocol)
       .then(({ title, image }) => {
-        updateLinkFromMetadataApi({ url: urlWithProtocol, title, icon: image });
+        if (title && image) {
+          return updateLinkFromMetadataApi({
+            url: urlWithProtocol,
+            title,
+            icon: image,
+          });
+        }
+
+        // This is necessary since the API is returning empty image/title fields sometimes.
```

Comment:
Could we fix the API instead? This seems a bit unstable.

Reply:
I think that is related to the `PHP` part, right? Should I create an issue for it?

Reply:
Ideally it would be fixed in the same issue here. Feel free to tag in @miina or @swissspidy to help out.

Reply:
Awesome! Thanks for the heads up. 

cc @miina @swissspidy can you pls help me achieve this improvement? Do you know why API returns empty fields after persist an invalid link?

**Steps to Reproduce**

1. Paste an invalid LINK in the `Web address` field and wait for the API to return the expected 404 error.
2. Paste the same link again in `Web address` field, now the API will return 200 with the following response:
![Screen Shot 2020-03-31 at 18 45 43](https://user-images.githubusercontent.com/1680157/78078207-1bf0de80-7380-11ea-98d4-1701b2a81e02.png)

I agree with @barklund,  for this case, the best way is to solve it in the back-end side.

Reply:
FYI, [this](https://github.com/google/web-stories-wp/blob/master/assets/src/edit-story/components/panels/link/index.js#L103) is the place where the problematic request happens.

Reply:
Sounds like bad caching somewhere. Should definitely be fixed in API.

Reply:
The API does it's best to get a title, image and description. But sometimes, it simply cant get these values. This is not an error. What exactly should be returned if these values are empty. A error object and a status of 404?

Reply:
@spacedmonkey This is what is happening when we request metadata from an invalid URL (Step 1 in my example above). So yeah, for consistency, an error object with status 404 is enough. Thoughts?

Reply:
Invalid urls and empty head tags will return a 404 error. See this code https://github.com/google/web-stories-wp/blob/master/includes/REST_API/Link_Controller.php#L134-L150

What are being returned are the defaults, see https://github.com/google/web-stories-wp/blob/master/includes/REST_API/Link_Controller.php#L107-L115
Defaults are not an error...

Reply:
A 404 will result in catch statement being triggered.

Reply:
> But these defaults are returning from invalid URLs. Is it expected?

The PHP does check if the URL is valid. My tests I was getting error objects back. See
![Screenshot from 2020-03-31 23-46-01](https://user-images.githubusercontent.com/237508/78128245-80a64a80-740d-11ea-8054-4e8b6a817f59.png)

Reply:
@spacedmonkey that's what is happening if you use the same URL twice: 
![Kapture-2020-04-01-at-10 51 02](https://user-images.githubusercontent.com/1680157/78145121-e17e5480-7406-11ea-901b-5466d43c86fe.gif)

First, it returns an error for invalid URLs, second, it returns 200 with empty fields. Is it expected?

Reply:
No, it should not. The issue is that in case of erroring, the default data is saved to cache, and then that's returned instead of the original error information with the second call.

Will update the API not use cache in case of erroring. cc @swissspidy

Reply:
eccad5c

Reply:
It works! Thanks @miina